drm-howto/modeset-atomic.c

```diff
@@ -856,14 +877,15 @@ static void modeset_draw(int fd)
 	drmEventContext ev;
 	struct modeset_dev *iter;
 
-	/* Initial modeset on all outputs.
-	 * This may be a full-modeset, if needed.
-	 * Also, note that this is done with a blocking
-	 * call.
-	 */
-	for (iter = modeset_list; iter; iter = iter->next) {
-		modeset_atomic_modeset(fd, iter);
```

Comment:
As stated above, why removing this from here?

Reply:
Since I thought it was just a style change, I've decided to keep the commit that config the outputs in the main (as the original author did: https://github.com/dvdhrm/docs/blob/master/drm-howto/modeset-vsync.c#L446) to maintain a link with the other examples. Is there something wrong with that?

Reply:
I haven't checked closely, but I suspect this is what breaks the modeset when fbdev emulation is disabled. Can you check that ?

In any case, it's not nice to move code around in a commit with subject "Add comments to atomic API example" :-)

Reply:
> I haven't checked closely, but I suspect this is what breaks the modeset when fbdev emulation is disabled. Can you check that ?

I've just checked, but it keeps failing. The problem is that `modeset_atomic_modeset()` is doing an atomic commit without configuring framebuffer/CRTC properties, what makes it fail. Moving this piece of code back to `modeset_draw()` does not fix the problem.

> In any case, it's not nice to move code around in a commit with subject "Add comments to atomic API example" :-)

Yes, you're totally right. I'll pay attention to this in the future :+1:

Reply:
@leandrohrb hm, I'm pretty sure this worked, so perhaps some other change in your commit broke it? Have you tested https://github.com/ezequielgarcia/docs/commit/2eecbf9aece939571447eedf320fe38dffae055 alone?

Reply:
I've just tested with your commit (no modifications) and noticed that it doesn't work for 
`/dev/dri/card2`. In my case, that's the card that is available when we have `CONFIG_DRM_BOCHS=y`.

Reply:
Hm, I have tested mostly with virtgpu.